cluster/kubernetes/release.go

```diff
@@ -49,16 +58,63 @@ func (c *Kubectl) connectArgs() []string {
 	return args
 }
 
-func (c *Kubectl) kubectlCommand(args ...string) *exec.Cmd {
-	cmd := exec.Command(c.exe, append(c.connectArgs(), args...)...)
-	cmd.Stdout = c.stdout
-	cmd.Stderr = c.stderr
-	return cmd
+func (c *Kubectl) execute(logger log.Logger, errs cluster.SyncError) {
+	defer c.changeSet.clear()
+
+	for _, cmd := range cmds {
+		defaultSet := newExecuteSet(cmd, "--namespace", "default")
+		otherSet := newExecuteSet(cmd)
+
+		for _, obj := range c.objs[cmd] {
+			set := defaultSet
+			if !obj.hasDefaultNamespace() {
+				set = otherSet
+			}
+			set.stage(obj)
+		}
+
+		c.exec(logger, defaultSet, cmd, errs)
+		c.exec(logger, otherSet, cmd, errs)
+	}
+}
+
+func (c *Kubectl) exec(logger log.Logger, s *executeSet, cmd string, errs cluster.SyncError) {
+	if len(s.objs) == 0 {
+		return
+	}
+	// Attempt to apply everything in s at once, else fallback to applying one at a time.
+	if err := c.doCommand(logger, s.rw, s.cmd...); err != nil {
```

Comment:
I anticipate that it'd be useful to keep count of syncs that have to fall back -- either a separate counter, or as a label of the sync counter.